### DISCUSSION TITLE
How can I bind this event to the Viewmodel in MVVM and obtain its Tag.

### DISCUSSION THREAD

**Discussion by wosiyourfather** [AvaloniaUI/Avalonia#12569] [Q&A]
How can I bind this event to the Viewmodel in MVVM and obtain its Tag.

**Comment by wosiyourfather** [comment]
I am currently like this <DockPanel PointerPressed="Op_PointerPressed" Tag="{Binding .}" Grid.Column="0">
							<Label Classes="Name"  Foreground="#5DE4FD"  Name="name" Content="{Binding name}"></Label>
							</DockPanel>

**Comment by wosiyourfather** [comment]
![3](https://github.com/AvaloniaUI/Avalonia/assets/53590491/f16517a8-0f4e-483e-a528-1dc1a7fc9c68)

**Comment by wosiyourfather** [comment]
I don't want his events to appear in the. cs file

**Comment by stevemonaco** [comment]
`Avalonia.Xaml.Behaviors` allows you to combine `EventTriggerBehavior` and `InvokeCommandAction` to do this. [See this sample.](https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/master/samples/BehaviorsTestApplication/Views/Pages/InvokeCommandActionView.axaml)

The syntax is quite verbose. I prefer subscribing to the event, handling it in code-behind, and invoking a VM command from code-behind.

**Reply by wosiyourfather** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/6c8ccd78-35f6-4a4f-a699-f2187668bd76)
Can you help me see what the reason is

**Reply by stevemonaco** [reply]
It's not built-in. You need to install the `Avalonia.Xaml.Behaviors` package from NuGet, if you haven't already.

**Reply by wosiyourfather** [reply]
![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/fbce84f8-b8e6-46da-8d29-09386caaf8e4)
Yes, it is, but this error was reported

**Reply by stevemonaco** [reply]
That appears to be something specific to how your project is arranged, not specific to behaviors. Is `ImportProjectsTable` a named control in the same XAML file? It seems that way because you're using a path binding through a `DataContext`. If this is the case, look at [these docs](https://docs.avaloniaui.net/docs/data-binding/binding-to-controls#binding-to-a-named-control) for how to bind to a named control. If you're using `CompiledBinding`s, the syntax requires a cast, too.

**Reply by wosiyourfather** [reply]
I am located in ItemsControl, how to find the datacontext of the vm

**Reply by stevemonaco** [reply]
Name the `ItemsControl`. eg. `<ItemsControl x:Name="items">`

If you want to bind to the `DataContext` it is assigned, then do something like:
```xml
<ia:InvokeCommandAction Command="{Binding #items.DataContext.Op_PointerPressedCommand}" CommandParameter="{Binding}" />
```

The `CommandParameter` in this case is the "current item VM" that's being displayed. You could pass a `Tag` instead, I guess, but I don't understand your logic here.

**Reply by wosiyourfather** [reply]
First of all, thank you, but through your method, did this issue occur due to incorrect naming in my viewmodel

![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/365063ca-0f9c-4494-afe8-2baea1007986)
![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/387c3ab6-ee2b-4059-b3d1-14cbd437c6b7)

**Comment by wosiyourfather** [comment]
![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/de8fa5f8-bd6f-48d3-8e4d-b2fc5ad90e29)
First of all, thank you, but through your method, did this issue occur due to incorrect naming in my viewmodel
![image](https://github.com/AvaloniaUI/Avalonia/assets/53590491/cea2d4c0-29aa-4b8e-a6c0-c1f7ce259b3a)

**Reply by stevemonaco** [reply]
There are a few potential issues here:

1. Your `Op_PointerPressedCommand` needs to take a parameter of the type you're passing in via `CommandParameter`.
2. It's a method and not an `ICommand` (eg. `ReactiveCommand` or `RelayCommand` in some MVVM frameworks). Avalonia has some support for binding to methods, but I'm not sure if that works with `InvokeCommandAction`. I've never tried.

I suggest you do some prototyping in a fresh project. MVVM is much different than how code is usually "organized" in WinForms.
